### PR TITLE
refactor: route browser tools through core APIs

### PR DESCRIPTION
**Pull request by shadowfax92**
## Summary
- add internal browser core page/session, observation, snapshot, and input compatibility paths behind the existing Browser facade
- move browser tool definitions under `apps/server/src/tools/browser` while preserving public tool names and schemas
- prune unused legacy browser helpers and add test coverage for the new browser tool boundary

## Verification
- `bun run --filter @browseros/server typecheck`
- `bun --env-file=apps/server/.env.development test apps/server/tests/index.test.ts apps/server/tests/tools/navigation.test.ts apps/server/tests/tools/observation.test.ts apps/server/tests/tools/input.test.ts apps/server/tests/tools/dom.test.ts apps/server/tests/tools/bookmarks.test.ts apps/server/tests/tools/history.test.ts apps/server/tests/tools/windows.test.ts apps/server/tests/tools/tab-groups.test.ts apps/server/tests/tools/page-actions.test.ts apps/server/tests/api/screencast-manager.test.ts`
- `bun run --filter @browseros/server test:tools`
- `bun test apps/server/src/browser/core/snapshot/diff.test.ts apps/server/src/browser/core/snapshot/render.test.ts apps/server/src/browser/core/observer/resolve.test.ts`

### CODE REVIEW

**Comment by shadowfax92** [comment]
Addressed the Greptile feedback in a1530347: removed the disabled wait_for tool/import/prompt reference, fixed the PageManager.newPage retry delay, and delegated Browser facade navigation through core Navigation. Latest CI is green.

File: packages/browseros-agent/apps/server/src/browser/core/pages.ts [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    let tab: TabInfo | undefined
+    for (let attempt = 0; attempt < 30; attempt++) {
+      try {
+        tab = (await this.cdp.Browser.getTabInfo({ tabId })).tab as TabInfo
+        if (!tab.isLoading || tab.loadProgress >= 1) break
+      } catch {
+        await delay(100)
+      }
+      await delay(100)
+    }
```
[the remark this answers is not in the document]
Comment (shadowfax92) [reply packages/browseros-agent/apps/server/src/browser/core/pages.ts:210]: Fixed in a1530347 by removing the catch-path delay, leaving one 100ms delay per retry attempt.
